docs/howtoguides/pro_in_airgapped.md

```diff
@@ -0,0 +1,102 @@
+```bash
+/usr/lib/apt/apt-helper download-file https://bearer:[ESM_INFRA_RESOURCE_TOKEN]@esm.ubuntu.com/ubuntu/ /tmp/check-esm-resource-token.txt
+```
+Response must be `200`. `401 Unauthorized` shows that the resource token or the URL is wrong. Please note that the specifics of setting up mirrors may be different depending on the product, but the bearer authentication will be identical in all cases, so refer to the command above to distinguish between airgapped and mirror problems.
+
+## Set up mirrors
+If you have experience in setting up mirrors, this section is optional for you. However, if you want to get started with the simplest option of setting up mirrors and have no prior experience, keep reading.
+
+Suppose you want to set up mirrors for esm-infra. Then, the steps to set up a mirror would be:
+1. Install `apt-mirror`: \
+`sudo apt install apt-mirror`
+2. Fetch the resource token for `esm-infra` (as per the section above).
+3. Change the `/etc/apt/mirror.list` file to the following contents:
+	```
+	set nthreads     20
+	set _tilde 0
+
+	deb https://bearer:[ESM_INFRA_RESOURCE_TOKEN]@esm.ubuntu.com/infra/ubuntu/ jammy-infra-updates main
+	deb https://bearer:[ESM_INFRA_RESOURCE_TOKEN]@esm.ubuntu.com/infra/ubuntu/ jammy-infra-security main
+
+	clean http://archive.ubuntu.com/ubuntu
+	```
+4. Run `apt-mirror`: \
+`sudo -u apt-mirror apt-mirror`
+5. Serve the `esm-infra` mirror on port `9090` (will be `[ESM_INFRA_MIRROR_URL]` in the next section): \
+`nohup sh -c 'python3 -m http.server --directory /var/spool/apt-mirror/mirror/esm.ubuntu.com/infra/ 9090 > /dev/null 2>&1 &'`
+
+
+## Get configuration to run the server
+The prerequisites for this step are:
+1. Mirrors of needed resources are set up on the airgapped machine
+2. Mirrors of needed resources are served on some port on the airgapped machine
+
+We need to create such a configuration for our server so that it points to the local mirror server on the airgapped machine, not the Canonical one. For example, if we want to use `esm-infra` in the airgapped environment, create the `02-overridden.yml` file on the non-airgapped machine with the following contents:
+```yaml
+[YOUR_CONTRACT_TOKEN]:
+  esm-infra:
+    directives:
+      aptURL: [ESM_INFRA_MIRROR_URL]
+```
+Run `ua-airgapped` for the new config in non-airgapped environment: `cat 02-overridden.yml | ua-airgapped > 02-server-ready.yml`
+
+`02-server-ready.yml` is our final configuration to run the server.
+
+## Running the server and final tweaks
+Run the following commands in the airgapped environment:
+1. Run the server: `contracts-airgapped --input=./02-server-ready.yml`
+2. Change the `contract_url` setting in `uaclient.conf` to point to the server: `http://1.2.3.4:8484`. Here `1.2.3.4` is the IP address of the airgapped machine.
+3. Run `sudo pro refresh` to check everything works fine.
```

Comment:
I think we can ask the user to also run `pro attach` to see that the services are indeed enabled

Reply:
Done.